src/pages/OnboardingWork/BaseOnboardingWork.tsx

```diff
@@ -22,14 +22,18 @@ import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
 import INPUT_IDS from '@src/types/form/WorkForm';
-import type {BaseOnboardingWorkOnyxProps, BaseOnboardingWorkProps} from './types';
+import type {BaseOnboardingWorkProps} from './types';
 
-function BaseOnboardingWork({shouldUseNativeStyles, onboardingPurposeSelected, onboardingPolicyID, route}: BaseOnboardingWorkProps) {
+function BaseOnboardingWork({shouldUseNativeStyles, route}: BaseOnboardingWorkProps) {
     const styles = useThemeStyles();
     const {translate} = useLocalize();
     const {onboardingIsMediumOrLargerScreenWidth} = useResponsiveLayout();
     const {inputCallbackRef} = useAutoFocusInput();
 
```

Comment:
Everything else helped a lot, there's only one missing issue with route/navigation:

- The Purpose screen shows up for an instant before loading Work. This is not a huge deal, but...
- After submitting business name, there's a flash navigation to the next screen but it goes back to Work, requiring the user to submit the name twice. What could be causing this navigation issue? I'm thinking that maybe it's fixed if we find a way to not render Purpose automatically, but actually start with Work when the `signupQualifier` is `vsb`. Thoughts?

https://github.com/user-attachments/assets/222fc37c-2e44-4eaf-8614-5fcde32ff74c

Reply:
Oh, interesting... Not sure what we can do about that, let's see if @tgolen has a suggestion (video above, Tim, any idea why we're getting a weird navigation issue that causes the need to submit twice, any suggestion for where to look?).

Reply:
I chatted a little with Carlos about this yesterday, so I wanted to post my findings here. As I was looking into this I found that [this specific redirect](https://github.com/Expensify/App/blob/e8b7582bb68218358d5402e039def8f618899957/src/pages/OnboardingPurpose/BaseOnboardingPurpose.tsx#L57)

- Happens 3 times immediately when dropped onto the page (yikes)
- Is also the same redirect that keeps you on the `onboard/work` form and forces you to submit it twice

I think it's then just a matter of figuring out what's going on with `showBusinessModal`.

Reply:
Gandalf summoned me here! (always wanted to say this hahah)

```ts
    if (showBusinessModal) {
        Navigation.navigate(ROUTES.ONBOARDING_WORK.getRoute(route.params?.backTo));
    }
```

We can't have the above code because it will be triggered each time the component is re-rendered, a better approach would be to put this logic inside `useEffect` with just one dependency (`showBusinessModal`), I think this way it should just work.

Reply:
I'm leaning towards a better long-run approach, we should better calculate initial path of onboarding flow, because if I understand correctly `ONBOARDING_WORK` should be displayed instead of `ONBOARDING_PURPOSE` if this condition is met:
```ts
    const vsbOnboarding = onboardingValues && 'signupQualifier' in onboardingValues && onboardingValues.signupQualifier === CONST.ONBOARDING_SIGNUP_QUALIFIERS.VSB;
```

I prefer adjusting it in `src/libs/actions/Welcome/OnboardingFlow.ts`, for example in `getOnboardingInitialPath` function
```ts
function getOnboardingInitialPath(): string {
    const state = getStateFromPath(onboardingInitialPath, linkingConfig.config);
    if (state?.routes?.at(-1)?.name !== NAVIGATORS.ONBOARDING_MODAL_NAVIGATOR) {
        return `/${ROUTES.ONBOARDING_ROOT.route}`;
    }

    return onboardingInitialPath;
}
```

Reply:
Hi everyone, I'm back after dealing with a networking issue that we couldn't solve and forced me to wipe out and reisntall everything in my computer. I'll look at all of this now and let you know if I have any questions, thanks everyone!

Reply:
Just implemented and tested and it works great! Checking if we can skip the refactoring of `AuthScreens` to merge this PR quicker, will report back.